stdlib/public/runtime/HeapObject.cpp

```diff
@@ -116,6 +154,20 @@ static HeapObject *_swift_tryRetain_(HeapObject *object)
   return _ ## name ## _ args; \
 } while(0)
 
+// SWIFT_REFCOUNT_CC functions make the call to the "might be swizzled" path
+// through an adapter marked noinline and with the refcount CC. This allows the
+// fast path to avoid pushing a stack frame. Without this adapter, clang emits
+// code that pushes a stack frame right away, then does the fast path or slow
+// path.
+#define CALL_IMPL_SWIFT_REFCOUNT_CC(name, args)                                          \
+  do {                                                                                   \
+    if (SWIFT_UNLIKELY(                                                                  \
+            _swift_enableSwizzlingOfAllocationAndRefCountingFunctions_forInstrumentsOnly \
```

Comment:
FWIW Instruments never ended up adopting this, in case that's useful info. They pointed out that exposing a C++ atomic this way risks ABI issues, and asked for a function to call instead.